cloudevents.md

```diff
@@ -1734,4 +1808,758 @@ This *approval.finished* event contains the result of the approval.
+    "stage": "staging",
+    "service": "carts",
+    "labels": {
+      "testid": "12345",
+      "buildnr": "build17",
+      "runby": "JohnDoe"
+    },
+  }
+}
+```
+
+</p>
+</details>
+
+([&uarr; up to index](#keptn-cloud-events))
+
+## Remediation Status Changed
+
+The *remediation.status.changed* event is sent when a remediation action was triggered but there are further remediation actions available.
+
+### type
+```json
+"type": "sh.keptn.event.remediation.status.changed"
+```
+
+### data
+```json
+"RemediationStatusChangedEventData": {
+  "required": [
+    "remediation",
+    "project",
+    "stage",
+    "service",
+  ],
+  "properties": {
+    "remediation": {
+      "required": [
+        "status",
+        "result",
+      ],
+      "status": { // Enum: succeeded, errored, unknown
+        "type": "string" 
+      },
+      "result": {
+        "required": [
+          "actionIndex",
+          "actionName",
+        ],
+        "actionIndex": {
+          "type": "integer"
+        },      
+        "actionName": {
+          "type": "string"
+        },
+      },
+      "type": "object"
+    },
+    "project": {
+      "type": "string"
+    },
+    "stage": {
+      "type": "string"
+    },
+    "service": {
+      "type": "string"
+    },
+    "labels": {
+      "patternProperties": {
+        ".*": {
+          "type": "string"
+        }
+      },
+      "type": "object"
+    },
+  },
+  "additionalProperties": false,
+  "type": "object"
+}
+```
+
+### Example
+<details><summary>Example of sh.keptn.event.remediation.status.changed</summary>
+<p>
+
+```json
+{
+  "type": "sh.keptn.event.remediation.status.changed",
+  "specversion": "0.2",
+  "source": "https://github.com/keptn/keptn/remediation-service",
+  "id": "ggb878d3-03c0-4e8f-bc3f-454bc1b3d888",
+  "time": "2019-06-07T07:02:15.64489Z",
+  "contenttype": "application/json",
+  "shkeptncontext": "08735340-6f9e-4b32-97ff-3b6c292bc509",
+  "triggerid": "f2b878d3-03c0-4e8f-bc3f-454bc1b3d79d",
+  "data": {
+    "remediation": {
+      "status": "succeeded",
+      "result": {
+        "finishedActionIndex": 0,
+        "finishedActionName": "scaling",
+      }
+    },
+    "project": "sockshop",
+    "stage": "staging",
+    "service": "carts",
+    "labels": {
+      "testid": "12345",
+      "buildnr": "build17",
+      "runby": "JohnDoe"
+    },
+  }
+}
+```
+
+</p>
+</details>
+
+([&uarr; up to index](#keptn-cloud-events))
+
+## Remediation Finished
+
+The *remediation.finished* event is sent when a remediation action is finished
+
+### type
+```json
+"type": "sh.keptn.event.remediation.finished"
+```
+
+### data
+```json
+"RemediationFinishedEventData": {
+  "required": [
+    "remediation",
+    "project",
+    "stage",
+    "service",
+  ],
+  "properties": {
+    "remediation": {
+      "required": [
+        "status",
+        "result",
+      ],
+      "status": { // Enum: succeeded, errored, unknown
+        "type": "string" 
+      },
+      "result": {
```

Comment:
According to Issue [#1849](https://github.com/keptn/keptn/issues/1849), it should also possible to include arbitrary strings within the result (e.g. "remediation file not configured"). Should we add an additional field (e.g. `message`) here, or should we remove the enum restriction?

Reply:
I would propose to introduce an additional `message` field. Otherwise, the event receiver would need to correctly interpret the result message. This can lead to misinterpretation.
The same is true for the `status` field. 

Should we add: `statusMessage`, `resultMessage` (which are optional)?

@agrimmer What is your opinion on that?

Reply:
I would prefer a "message" field because this would allow us to automatically parse and evaluate the "status".

Reply:
I would prefer introducing a new field "message". This would allow us to use the "result" field for the evaluation result.